src/lib/utils.ts

```diff
@@ -34,3 +37,47 @@ export const loadScriptTag = async (srcValue: string): Promise<void> => {
 
   await loadedPromise
 }
+
+export const getUserVariables = (): {userId: string, courseId: string, instance: string} | null => {
+  const userId = '2'
+  const courseId = window.M?.cfg.courseId
+  const instance = window.location.host
+  if (userId === undefined || courseId === undefined || instance === undefined) {
+    return null
+  }
+  return {
+    userId,
+    courseId,
+    instance
+  }
+}
+
+function getVariantMappings(host: string, courseid: string): string {
+  const mappingData = variantMappings as any
+  const maybeDataMapping = mappingData.default[host]?.courses[courseid]
```

Comment:
Just curious: What's the `.default` for?

Reply:
It was just the way the object was structured when imported in this manner - the whole thing was wrapped in a "default" field. I couldn't find a way to create the object otherwise. 🤷‍♂️ (sry not a fun answer)

Reply:
Oh -- I see. I think to avoid this you can do your import as:

```javascript
import variantMappings from '../../data/variant-mapping.json'
```

Then this code just becomes:


```javascript
  const mappingData = variantMappings as any
  const maybeDataMapping = mappingData[host]?.courses[courseid]
```

Reply:
Ahhhh 👏 nice